depthmapXcli/runmethods.cpp

```diff
@@ -53,6 +54,13 @@ namespace dm_runmethods
         return mgraph;
     }
 
+    Communicator* getCommunicator(const CommandLineParser &clp) {
+        if (clp.printProgress()) {
+            return new PrintCommunicator();
```

Comment:
Creating a communicator with `new` that never gets deleted? I know that this most likely get destroyed when the client program shuts down, but this is still a dirty resource leak. Should be assigned to a smart pointer.